include/rocksdb/compaction_filter.h

```diff
@@ -16,11 +16,15 @@
 
 #include "rocksdb/customizable.h"
 #include "rocksdb/rocksdb_namespace.h"
+#include "rocksdb/table_properties.h"
 #include "rocksdb/types.h"
 #include "rocksdb/wide_columns.h"
 
 namespace ROCKSDB_NAMESPACE {
 
+using TablePropertiesCollection =
+    std::unordered_map<std::string, std::shared_ptr<const TableProperties>>;
```

Comment:
This looks the same as the one in listener.h, so it should go in types.h

Reply:
Unfortunately, there can be a circular dependency issue where types.h needs to include  table_properties.h due to `std::shared_ptr<const TableProperties>` but TableProperties also needs `EntryType` from types.h due to 
```
 virtual Status AddUserKey(const Slice& key, const Slice& value,
                            EntryType /*type*/, SequenceNumber /*seq*/,
                            uint64_t /*file_size*/) 
```

Reply:
What if you forward declare TableProperties in types.h?

Reply:
Oh right - I forgot our forward declaration friend. Fixed.